llap-server/src/java/org/apache/hadoop/hive/llap/io/api/impl/LlapRecordReader.java

```diff
@@ -158,8 +167,11 @@ private LlapRecordReader(MapWork mapWork, JobConf job, FileSplit split,
     rbCtx = ctx != null ? ctx : LlapInputFormat.createFakeVrbCtx(mapWork);
 
     isAcidScan = AcidUtils.isFullAcidScan(jobConf);
-    TypeDescription schema = OrcInputFormat.getDesiredRowTypeDescr(
-        job, isAcidScan, Integer.MAX_VALUE);
+
+    String icebergOrcSchema = job.get(ColumnProjectionUtils.ICEBERG_ORC_SCHEMA_STRING);
```

Comment:
This is a little strange here. I would try to avoid using Iceberg specific stuff in LLAP packages.
Maybe a job config containing the requested schema?

Reply:
Yeah it's unfortunate. This is the equivalent of non-LLAP but vectorized ORC case at https://github.com/apache/hive/blob/master/ql/src/java/org/apache/hadoop/hive/ql/io/orc/VectorizedOrcInputFormat.java#L80

Hive sets IOConstants.SCHEMA_EVOLUTION_COLUMNS and SCHEMA_EVOLUTION_COLUMNS_TYPES during compile and pushes this down to execution. These are simple string representations of Hive types so not ORC specific, and relates to LOGICAL schema.
Iceberg - in order to support a broader schema evolution - produces the FILE schema based on file info and logical type info, and the result is an ORC TypeDescription instance. (See VectorizedReadUtils.handleIcebergProjection) I have found no easy way to transform this object back into Hive types, the conversion only exists in the other direction: OrcInputFormat.typeDescriptionsFromHiveTypeProperty()

Reply:
As discussed offline this is now renamed so that it less iceberg-specific.